src/types/value/contentRelationship.ts

```diff
@@ -22,18 +22,21 @@ export type ContentRelationshipField<
 	State extends FieldState = FieldState,
 > = State extends "empty"
 	? EmptyLinkField<typeof LinkType.Document>
-	: FilledContentRelationshipField<TypeEnum, LangEnum, DataInterface>
+	: Omit<
+			FilledContentRelationshipField<TypeEnum, LangEnum, DataInterface>,
+			"text" | "variant"
+		>
 
 /**
  * Links that refer to documents
  */
-export interface FilledContentRelationshipField<
+export type FilledContentRelationshipField<
 	TypeEnum = string,
 	LangEnum = string,
 	DataInterface extends
 		| Record<string, AnyRegularField | GroupField | SliceZone>
 		| unknown = unknown,
-> {
+> = WithLinkAdditionalData<{
```

Comment:
not sure i understand this one, we wrap FilledContentRelationshipField with WithLinkAdditionalData, adding those two additional props to it? i thought we didn't want this

Reply:
Yes, we want this because it's used everywhere internally to express what is a valid Link To Document even for Generic Link.

Reply:
sorry, i don't understand

for link the generic link, we wrap ContentRelationshipField with the additional data, so it's covered when we have a link to document via generic link

but why do we want this as well when it's a standalone content relationship field? when devs use this one it'll have the optional properties even though the model doesn't allow them

Reply:
i see now

filled contet rel. has the add. props
we remove them from content rel.
we add them again when we use content rel. inside the generic link